R/data_inventory_table.R

```diff
@@ -467,3 +467,128 @@ pt_data_inventory_notes <- function(bq = c("BQL", "BLQ"), drop_bql = FALSE, note
   )
   ans
 }
+
+
+#' Data inventory by covariate
+#'
+#' Generates a table showing numbers of subjects and observations, stratifying
+#' by multiple categorical covariates.
+#'
+#' @inheritParams pt_data_inventory
+#'
+#' @param cols data columns containing discrete data items for grouped data
+#' inventory summaries.
+#' @param level_width width of the `level` column, specified in `cm`.
+#'
+#' @examples
+#'
+#' data <- pmt_first
+#'
+#' tab <- pt_inventory_long(data, cols = c("FORMf", "SEXf", "RFf"))
+#'
+#' tab$data
+#'
+#' @seealso [pt_data_inventory()]
+#'
+#' @export
+pt_inventory_long <- function(data,
+                              cols,
+                              drop_miss = FALSE,
+                              table =  NULL,
+                              summarize_all = TRUE,
+                              all_name = "All data",
+                              dv_col = "DV",
+                              bq_col = find_bq_col(data),
+                              id_col = "ID",
+                              level_width = NULL) {
+
+  assert_that(is.data.frame(data))
+  assert_that(nrow(data) > 1)
+  assert_that(is.character(cols))
+  assert_that(length(cols) > 0)
+  assert_that(is.character(all_name))
+  assert_that(length(all_name)==1)
+
+  if(!missing(table)) {
+    assert_that(is.list(table))
+    assert_that(!is.data.frame(table))
+  }
+
+  data <- as.data.frame(data)
```

Comment:
Is there a point in this `as.data.frame` call given the `assert_that(is.data.frame(data))` guard upstream?

Reply:
Make sure we're dealing with a vanilla data frame rather than tibble.